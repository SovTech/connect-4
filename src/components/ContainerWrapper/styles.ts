import theme from '../../theme';
import styled from 'styled-components';

export const ContainerWrapperOuter = styled.div`
    display: flex;
    flex-direction: column;
    flex: 1;
    align-items: center;
    align-self: center;
    justify-content: center;
    min-height: calc(100vh - 30px);
`;

export const ContainerWrapperInner = styled.div`
    display: flex;
    flex-direction: column;
    flex: 1;
    align-items: center;
    align-self: center;
    justify-content: center;
    max-width: ${theme.maxContentWidth};
    margin-top: ${theme.headerHeight};
`;

export const VersionNumber = styled.div`
    color: ${theme.textColor};
`;