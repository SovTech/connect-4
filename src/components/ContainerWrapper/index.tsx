import * as React from 'react';
import { getPackageVersionNumber } from '../../utils';
import { ContainerWrapperInner, ContainerWrapperOuter, VersionNumber } from './styles';

type Props = {
  children: any;
}

const ContainerWrapper = (props: Props) =>
  (
    <ContainerWrapperOuter>
      <ContainerWrapperInner>
        {props.children}
      </ContainerWrapperInner>
      <VersionNumber>v{getPackageVersionNumber()}</VersionNumber>
    </ContainerWrapperOuter>
  );

export default ContainerWrapper;
