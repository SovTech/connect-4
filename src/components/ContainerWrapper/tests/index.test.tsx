import * as React from 'react';
import * as renderer from 'react-test-renderer';
import ContainerWrapper from '../index';

it('renders a Cell correctly', () => {
  const tree = renderer
    .create((
      <ContainerWrapper>
        <div>das kinders</div>
      </ContainerWrapper>))
    .toJSON();
  expect(tree).toMatchSnapshot();
});
