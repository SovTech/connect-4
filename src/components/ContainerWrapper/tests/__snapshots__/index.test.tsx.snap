// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a Cell correctly 1`] = `
<div
  className="sc-bdVaJa kFXOhx"
>
  <div
    className="sc-bwzfXH imUrkV"
  >
    <div>
      das kinders
    </div>
  </div>
  <div
    className="sc-htpNat bUJwTp"
  >
    v
    x.x.x
  </div>
</div>
`;
